### ISSUE TITLE
Options followed by empty Vecs are misparsed (or misseralized)

### ISSUE THREAD

**Issue by boustrophedon** [BurntSushi/rust-csv#110]
If we have a struct like so:

```rust
#[derive(Serialize, Deserialize)]
struct Record {
  x: i32,
  y: Option<i32>,
  z: Vec<i32>
}
```

Then if the vec is empty and the option is none, it is serialized as "1,". This fails to be deserialized because the reader expecting three things, but only sees 1. If you wrap the Vec is an Option<> then it works.

Full example: https://play.rust-lang.org/?gist=ebca1d1ef55cf4565e97c9a8a78fbed5&version=stable

**Comment by boustrophedon** [comment]
Actually, it might just be that empty vecs by themselves are bad?

**Comment by BurntSushi** [comment]
The serialization seems correct to me at least. If the vec is empty, then there is nothing to serialize.

The deserialization is also interesting, but not completely unsurprising to me. I think what you're requesting basically boils down to whether an empty vec and a non-existent vec should be treated the same. It seems like it it probably should. That is, a `Vec<T>` should succeed even if there are no fields to parse.

**Comment by boustrophedon** [comment]
Yeah, it makes sense to me that you should be able to deserialize an empty vec. It's a fairly easy workaround for now: wrap vecs in an Option.
